plugins/check-path/core-rs/src/paths.rs

```diff
@@ -0,0 +1,84 @@
+use js_sys::Reflect;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+use wasm_bindgen::prelude::*;
+
+enum NodeValue {
+    StringType(String),
+    ObjectType(JsValue),
+    None,
+}
+
+pub struct Paths {
+    store: Rc<RefCell<HashMap<String, Vec<String>>>>,
+}
+
+impl Paths {
+    pub fn new() -> Self {
+        Self {
+            store: Rc::new(RefCell::new(HashMap::new())),
+        }
+    }
+
+    pub fn parse(&self, data: JsValue) {
```

Comment:
What's the point of this function? Why not inline traverse into it, or remove it and make traverse pub?